tapsend/send.go

```diff
@@ -364,6 +372,46 @@ func ValidateInputs(inputCommitments tappsbt.InputCommitments,
 	return isFullValueSpend, nil
 }
 
+// ValidateScriptKeysUnique makes sure the outputs of a set of virtual packets
+// don't lead to collisions in and of the trees (e.g. two asset outputs with the
+// same script key in the same anchor output) or with exclusion proofs (e.g. two
+// asset outputs with the same script key in different anchor outputs).
+func ValidateScriptKeysUnique(packets []*tappsbt.VPacket) error {
+	// We'll keep track of all script keys (per top commitment key) that
+	// we've seen so far. Because the script key is the lower tree's
+	// commitment key, it must be unique per top commitment key.
+	scriptKeys := make(map[[32]byte]map[asset.SerializedKey]struct{})
+
+	for _, vPkt := range packets {
+		tapCommitmentKey, err := vPkt.TapCommitmentKey()
+		if err != nil {
+			return fmt.Errorf("unable to determine tap commitment "+
+				"key: %w", err)
+		}
+
+		perAssetMap, ok := scriptKeys[tapCommitmentKey]
+		if !ok {
+			perAssetMap = make(map[asset.SerializedKey]struct{})
+			scriptKeys[tapCommitmentKey] = perAssetMap
+		}
+
+		for idx := range vPkt.Outputs {
+			vOut := vPkt.Outputs[idx]
+
+			// Check if the script key has already been used in this
```

Comment:
Isn't it ok for us to have a script key duplicate _across_ Bitcoin transaction outputs?

Reply:
I think a duplicate script key for the same asset would be an issue, even without those assets being children of an asset split. If that occurred, the exclusion proof would be a proof of inclusion, but for a _different_ leaf vs. the existing model where the exclusion proof terminates in a nil leaf.

Reply:
Hmm yeh, I do wonder if there's a small change we can make the the exclusion proofs here to patch this portion.